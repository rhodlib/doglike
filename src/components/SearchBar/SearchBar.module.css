.SearchBar {
  width: 25%;
  height: 25px;
  min-width: 200px;
  margin-top: 0px;
  border-radius: 5px;
}
