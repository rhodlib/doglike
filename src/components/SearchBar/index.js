import React from "react";
import styles from "./SearchBar.module.css";
import PropTypes from 'prop-types';

export const SearchBar = ({ dogList = [], ...props }) => (
  <>
    <input className={styles.SearchBar} list="dogs" {...props} />
    <datalist id="dogs">
      {dogList.map(dog => (
        <option key={dog} value={dog} />
      ))}
    </datalist>
  </>
);

SearchBar.propTypes = {
  dogList: PropTypes.arrayOf(PropTypes.string),
}

export default SearchBar;
