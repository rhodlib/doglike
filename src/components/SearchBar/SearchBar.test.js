import React from "react";
import { render } from "@testing-library/react";
import SearchBar from "./index";

it("Matches snapshot", () => {
  const { asFragment } = render(<SearchBar/>);
  expect(asFragment()).toMatchSnapshot();
});

it("Matches snapshot", () => {
    const { asFragment } = render(<SearchBar dogList={["test"]} />);
    expect(asFragment()).toMatchSnapshot();
  });