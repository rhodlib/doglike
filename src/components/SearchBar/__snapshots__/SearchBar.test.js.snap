// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <input
    class="SearchBar"
    list="dogs"
  />
  <datalist
    id="dogs"
  />
</DocumentFragment>
`;

exports[`Matches snapshot 2`] = `
<DocumentFragment>
  <input
    class="SearchBar"
    list="dogs"
  />
  <datalist
    id="dogs"
  >
    <option
      value="test"
    />
  </datalist>
</DocumentFragment>
`;
